### ISSUE TITLE
Resolve variables from per-query tenant field

### ISSUE THREAD

**Issue by jotak** [hawkular/hawkular-grafana-datasource#90]
As discussed in #85 templating variables are not resolved when used in the per-query tenant field.
Also in such case, the tenant name should probably be mentioned in the chart legend for disambiguation.

**Comment by ljuaneda** [comment]
Is this ready to be released ?

**Comment by jotak** [comment]
Did you have a chance to try it before I release it?

**Comment by ljuaneda** [comment]
I've tried to pull your commit and build the docker image
To do that I have to modify the Dockerfile as it was pulling the last release from github.

Can you provide a quick tuto so that I can build my own img ?

**Comment by jotak** [comment]
Actually, I've just found a bug with the new code, I must fix it first...

**Comment by ljuaneda** [comment]
ok thanks, I'd like to test that but I'm not sure I'm building the img the right way
a makefile could help anyway...

**Comment by jotak** [comment]
@ljuaneda I'm adding some doc about that: cf https://github.com/hawkular/hawkular-grafana-datasource/pull/92

**Comment by jotak** [comment]
(but it requires you to have dev environment, that is nodejs/npm/grunt)

**Comment by ljuaneda** [comment]
I have built an image following your doc, but something goes wrong using the resulting docker image:

``` bash
[wnradm@SIY05TY002:~]$ oc -n metrics-grafana logs -f hawkular-grafana-23-bbtpw
t=2017-09-18T12:00:22+0000 lvl=info msg="Starting Grafana" logger=main version=4.1.2 commit=v4.1.2 compiled=2017-02-13T12:13:31+0000
t=2017-09-18T12:00:22+0000 lvl=info msg="Config loaded from" logger=settings file=/usr/share/grafana/conf/defaults.ini
t=2017-09-18T12:00:22+0000 lvl=info msg="Config loaded from" logger=settings file=/etc/grafana/grafana.ini
t=2017-09-18T12:00:22+0000 lvl=info msg="Config overriden from command line" logger=settings arg="default.paths.data=/var/lib/grafana/data"
t=2017-09-18T12:00:22+0000 lvl=info msg="Config overriden from command line" logger=settings arg="default.paths.logs=/var/log/grafana"
t=2017-09-18T12:00:22+0000 lvl=info msg="Config overriden from command line" logger=settings arg="default.paths.plugins=/var/lib/grafana/plugins"
t=2017-09-18T12:00:22+0000 lvl=info msg="Config overriden from Environment variable" logger=settings var="GF_PATHS_DATA=/var/lib/grafana/data"
t=2017-09-18T12:00:22+0000 lvl=info msg="Config overriden from Environment variable" logger=settings var="GF_SERVER_ROOT_URL=http://metrics-grafana.infra.osev3-qt9-infra.hpmetier.sf.intra.laposte.fr"
t=2017-09-18T12:00:22+0000 lvl=info msg="Config overriden from Environment variable" logger=settings var="GF_DATABASE_TYPE=mysql"
t=2017-09-18T12:00:22+0000 lvl=info msg="Config overriden from Environment variable" logger=settings var="GF_DATABASE_HOST=mysql:3306"
t=2017-09-18T12:00:22+0000 lvl=info msg="Config overriden from Environment variable" logger=settings var="GF_DATABASE_NAME=grafana"
t=2017-09-18T12:00:22+0000 lvl=info msg="Config overriden from Environment variable" logger=settings var="GF_DATABASE_USER=admin"
t=2017-09-18T12:00:22+0000 lvl=info msg="Config overriden from Environment variable" logger=settings var="GF_DATABASE_PASSWORD=*********"
t=2017-09-18T12:00:22+0000 lvl=info msg="Path Home" logger=settings path=/usr/share/grafana
t=2017-09-18T12:00:22+0000 lvl=info msg="Path Data" logger=settings path=/var/lib/grafana/data
t=2017-09-18T12:00:22+0000 lvl=info msg="Path Logs" logger=settings path=/var/log/grafana
t=2017-09-18T12:00:22+0000 lvl=info msg="Path Plugins" logger=settings path=/var/lib/grafana/plugins
t=2017-09-18T12:00:22+0000 lvl=info msg="Initializing DB" logger=sqlstore dbtype=mysql
t=2017-09-18T12:00:22+0000 lvl=info msg="Starting DB migration" logger=migrator
t=2017-09-18T12:00:22+0000 lvl=info msg="Starting plugin search" logger=plugins
t=2017-09-18T12:00:22+0000 lvl=info msg="Registering plugin" logger=plugins name=Hawkular
t=2017-09-18T12:00:22+0000 lvl=info msg="Initializing Alerting" logger=alerting.engine
t=2017-09-18T12:00:22+0000 lvl=info msg="Initializing CleanUpService" logger=cleanup
t=2017-09-18T12:00:22+0000 lvl=info msg="Initializing HTTP Server" logger=server address=0.0.0.0:3000 protocol=http subUrl=
t=2017-09-18T12:00:51+0000 lvl=info msg="Request Completed" logger=context userId=0 orgId=0 uname= method=GET path=/ status=302 remote_addr=220.4.96.10 time_ms=24ns size=29
t=2017-09-18T12:00:51+0000 lvl=info msg="Request Completed" logger=context userId=0 orgId=0 uname= method=GET path=/login status=302 remote_addr=220.4.96.10 time_ms=2ns size=24
t=2017-09-18T12:00:53+0000 lvl=info msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET path=/public/plugins/hawkular-datasource/img/hawkular_logo.svg status=404 remote_addr=220.4.96.10 time_ms=2ns size=13305
t=2017-09-18T12:01:02+0000 lvl=info msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET path=/public/plugins/hawkular-datasource/img/hawkular_logo.svg status=404 remote_addr=220.4.96.10 time_ms=3ns size=13305
t=2017-09-18T12:01:05+0000 lvl=info msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET path=/public/plugins/hawkular-datasource/module.js status=404 remote_addr=220.4.96.10 time_ms=3ns size=13305
```

Il looks like grafana plugin isn't correctly installed...
``` bash
[wnradm@SIY05TY002:~]$ oc -n metrics-grafana rsh hawkular-grafana-23-bbtpw ls -l /var/lib/grafana/plugins
total 0
drwxrwxr-x. 3 root root 49 Sep 15 13:53 hawkular-datasource
[wnradm@SIY05TY002:~]$ oc -n metrics-grafana rsh hawkular-grafana-23-bbtpw ls -l /var/lib/grafana/plugins/hawkular-datasource
total 0
drwxrwxr-x. 8 root root 256 Sep 15 13:53 hawkular-grafana-datasource-release
[wnradm@SIY05TY002:~]$ oc -n metrics-grafana rsh hawkular-grafana-23-bbtpw ls -l /var/lib/grafana/plugins/hawkular-datasource
total 0
drwxrwxr-x. 8 root root 256 Sep 15 13:53 hawkular-grafana-datasource-release
[wnradm@SIY05TY002:~]$ oc -n metrics-grafana rsh hawkular-grafana-23-bbtpw ls -l /var/lib/grafana/plugins/hawkular-datasource/hawkular-grafana-datasource-release
total 48
-rw-rw-r--. 1 root root  2454 Sep 12 12:39 Gruntfile.js
-rw-rw-r--. 1 root root 11323 Sep 12 12:39 LICENSE
-rw-rw-r--. 1 root root  9931 Sep 12 12:39 README.md
-rw-rw-r--. 1 root root  1462 Sep 12 12:39 RELEASE.adoc
drwxrwxr-x. 2 root root    43 Sep 12 12:39 dashboards
drwxrwxr-x. 2 root root    61 Sep 12 12:39 docker
drwxrwxr-x. 3 root root    20 Sep 12 12:39 docs
drwxrwxr-x. 2 root root    88 Sep 12 12:39 openshift
-rw-rw-r--. 1 root root  1162 Sep 12 12:39 package.json
-rw-rw-r--. 1 root root   718 Sep 12 12:39 plugin.json
-rwxrwxr-x. 1 root root   170 Sep 12 12:39 release.sh
drwxrwxr-x. 2 root root  4096 Sep 12 12:39 spec
drwxrwxr-x. 5 root root   284 Sep 12 12:39 src
```

**Comment by jotak** [comment]
That's weird, your plugin path is `/var/lib/grafana/plugins/hawkular-datasource/hawkular-grafana-datasource-release` whereas I would just expect `/var/lib/grafana/plugins/hawkular-datasource`

Not sure where this `hawkular-grafana-datasource-release` comes from, it looks like something brought by the other dockerfile (not the new Dockerfile.dev).

Can you make sure you used Dockerfile.dev (with command line argument `-f` passed to docker build)?

**Comment by jotak** [comment]
And can you also make sure that your `dist` directory doesn't contain this `hawkular-grafana-datasource-release`? (I don't know what it would, but it could be the reason)

**Comment by ljuaneda** [comment]
Sorry, my mistake...
I've rebuilt again, it is working just fine !!!

**Comment by jotak** [comment]
Cool!
I don't know if you noticed, I opened another PR here [ #93 ] with a fixed issue (metrics and tags suggestion with templated tenants were showing empty results). I hope that with that one, we'll be good

**Comment by jotak** [comment]
Just released 1.1.1, which addresses the remaining issues

**Comment by jotak** [comment]
(And dockerhub registry has been updated too)
